src/pages/home/ReportScreen.tsx

```diff
@@ -384,7 +384,7 @@ function ReportScreen({route, currentReportID = '', navigation}: ReportScreenPro
         !isCurrentReportLoadedFromOnyx ||
         // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
         (deleteTransactionNavigateBackUrl && ReportUtils.getReportIDFromLink(deleteTransactionNavigateBackUrl) === report?.reportID) ||
-        isLoading;
+        ((!ReportUtils.isUserCreatedPolicyRoom(report) || !reportMetadata.isOptimisticReport) && isLoading);
```

Comment:
Hmm now that I think about it this feels a bit hacky. Is there any reason to specifically apply this to policy rooms?

I've also noticed whilst testing that `isLoadingApp` gets set to true _only_ when the mode is switched to `Most recent` (not to `#focus`), which makes me think that we may have actually missed the true RCA here. I think `isLoadingApp` is either being incorrectly set when the focus mode is switched, or this is an unexpected side-effect of it.

Reply:
A similar behaviour happens on any existing report, where if you switch specifically from `#focus` to `Most recent`, only the most recent message loads, the rest are in a loading state.

Reply:
> I just want to ensure this change doesn’t unintentionally affect other report types I might not have context on. It’s important to verify these also work correctly with the new change.

Makes sense, limiting the scope can reduce the chance for regressions. Are we okay to just focus on rooms here @lakchote, or do you think we should fix other reports (e.g. group chats)?

> However, for existing reports, we show the loading indicator because we don’t know “how many actions are missing”

Okay, just gonna check [on Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1734424263668709) regarding the expected behaviour there to make sure.

Reply:
I'd say we need to fix the whole issue as a whole and fix this for other reports.
If this introduces potential side effects/regressions, then we'll need to fix it and that would indicate the RCA of the solution here wasn't the correct one. I'd prefer to have solved the root cause and not a possibly hack that works for certain report types.

Reply:
Okay gotcha - let's do it for all report types, then - unless you disagree @truph01.